packages/runtime/container-runtime/src/orderedClientElection.ts

```diff
@@ -288,28 +338,44 @@ export class OrderedClientElection
                 });
             } else if (initialClient !== undefined && !isEligibleFn(initialClient)) {
                 // Initially elected client is ineligible, so elect next eligible client.
-                initialClient = this.findFirstEligibleClient(initialClient);
+                initialClient = initialParent = this.findFirstEligibleParent(initialParent);
                 logger.sendErrorEvent({
                     eventName: "InitialElectedClientIneligible",
                     electionSequenceNumber: initialState.electionSequenceNumber,
                     expectedClientId: initialState.electedClientId,
                     electedClientId: initialClient?.clientId,
                 });
             }
+            this._electedParent = initialParent;
             this._electedClient = initialClient;
             this._electionSequenceNumber = initialState.electionSequenceNumber;
         }
     }
 
     /** Tries changing the elected client, raising an event if it is different. */
     private tryElectingClient(client: ILinkedClient | undefined, sequenceNumber: number): void {
-        this._electionSequenceNumber = sequenceNumber;
-        if (this._electedClient === client) {
-            return;
-        }
+        let change = false;
+        const isSummarizerClient = client?.client.details.type === summarizerClientType;
         const prevClient = this._electedClient;
-        this._electedClient = client;
-        this.emit("election", client, sequenceNumber, prevClient);
+        if (this._electedClient !== client) {
```

Comment:
From the big comment above, this function will be called in step iii right. These are the sequence of steps:
1. electedParent = A, electedClient = A'.
2. `tryElectingClient` is called with client = B which is not a summarizer client.
3. This function will update both electedParent and electedClient to B.
But that should not happen right? It should only update electedParent. electedClient should be updated when A' leaves the quorum and this is called again. Is this understanding correct?

Reply:
This is a little hard to follow :-(. I like the big comment but I think we should add smaller comments from it in these functions as well so we can piece this info together.

Reply:
tryElectingClient should only get called if _electedClient is undefined or if the current _electedClient is not a summarizer client. It shouldn't be called to replace a summarizer client unless the summarizer client is leaving the quorum (i.e., in OrderedClientElection.removeClient).